app/src/main/java/com/weatherxm/util/NumberUtils.kt

```diff
@@ -1,35 +1,61 @@
 package com.weatherxm.util
 
 import android.icu.text.CompactDecimalFormat
-import android.icu.text.NumberFormat
-import com.weatherxm.util.Weather.EMPTY_VALUE
+import com.weatherxm.data.DECIMAL_FORMAT_TOKENS
+import com.weatherxm.ui.common.Contracts.EMPTY_VALUE
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
+import org.koin.core.qualifier.named
+import java.math.BigDecimal
 import java.math.RoundingMode
+import java.text.DecimalFormat
+import java.text.NumberFormat
 
 object NumberUtils : KoinComponent {
     private val compactDecimalFormat: CompactDecimalFormat by inject()
     private val numberFormat: NumberFormat by inject()
+    private val decimalFormat: DecimalFormat by inject(named(DECIMAL_FORMAT_TOKENS))
+
+    private const val DIVISOR_WEI_TO_ETH = "1000000000000000000"
+    private const val ETH_DECIMALS = 18
 
     fun compactNumber(number: Number?): String {
         return number?.let {
             compactDecimalFormat.format(number)
         } ?: EMPTY_VALUE
     }
 
-    fun formatNumber(number: Number?, maxDecimals: Int = 0): String {
+    fun formatNumber(number: Number?, decimals: Int = 0): String {
         return number?.let {
-            numberFormat.maximumFractionDigits = maxDecimals
+            numberFormat.minimumFractionDigits = decimals
+            numberFormat.maximumFractionDigits = decimals
             numberFormat.format(it)
         } ?: EMPTY_VALUE
     }
 
-    fun roundToDecimals(value: Number, decimals: Int = 1): Float {
-        return value.toFloat().toBigDecimal().setScale(decimals, RoundingMode.HALF_UP).toFloat()
+    fun formatTokens(amount: Float?): String {
+        return amount?.let {
+            formatTokens(it.toBigDecimal())
+        } ?: EMPTY_VALUE
+    }
+
+    fun formatTokens(amount: BigDecimal): String {
+        return decimalFormat.format(amount)
+    }
+
+    @Suppress("MagicNumber")
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/342#issuecomment-2388448877>#04ac52</a><span> - 10/02/2024, 12:12 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The weiToETH function doesn't handle potential overflow for very large Wei amounts, which could lead to incorrect ETH calculations or runtime exceptions.
</div>
<div id="fix">
<b>Fix:</b> Implement proper overflow checking and handling in the weiToETH function. Consider using BigInteger for intermediate calculations to avoid potential overflow.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -46,10 +46,15 @@ object NumberUtils : KoinComponent {
  @Suppress("MagicNumber")
  fun weiToETH(amount: BigDecimal): BigDecimal {
      /**
       * Mandatory otherwise we get a result of 0E-18 instead of 0
       */
      if (amount == BigDecimal.ZERO) {
          return BigDecimal.ZERO
      }
 -    return amount.divide(BigDecimal(DIVISOR_WEI_TO_ETH), ETH_DECIMALS, RoundingMode.HALF_UP)
 +    val divisor = BigDecimal(DIVISOR_WEI_TO_ETH)
 +    return try {
 +        amount.divide(divisor, ETH_DECIMALS, RoundingMode.HALF_UP)
 +    } catch (e: ArithmeticException) {
 +        Log.e("NumberUtils", "Error converting Wei to ETH: ${e.message}")
 +        BigDecimal.ZERO // or handle the error appropriately
 +    }
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged